tsdb/engine/tsm1/cache.go

```diff
@@ -180,37 +187,79 @@ func (c *Cache) Keys() []string {
 
 // Values returns a copy of all values, deduped and sorted, for the given key.
 func (c *Cache) Values(key string) Values {
```

Comment:
we'll have to look at it. Although I would say we probably want to have recommended settings smaller than that for when snapshots get taken